docs/streams.html

```diff
@@ -158,6 +161,27 @@
         </p>
         <br>
 
+        <h2><a id="streams_processing_guarantee" href="#streams_processing_guarantee">Processing Guarantees</a></h2>
+
+        <p>
+            In stream processing, one of the most frequently asked question is "does my stream processing system guarantee that each record is processed once and only once, even if some failures are encountered in the middle of processing?"
+            Failing to guarantee exactly-once stream processing is a deal-breaker for many applications that cannot tolerate any data-loss or date duplicates, and in that case a batch-oriented framework is usually used in addition
+            to the stream processing pipeline, known as the <a href="http://lambda-architecture.net/">Lambda Architecture</a>.
+            Prior to 0.11.0.0, Kafka only provides at-least-once delivery guarantees and hence any stream processing systems that leverages it as the backend storage could not guarantee end-to-end exactly-once semantics.
+            In fact, even for those stream processing systems that claim to support exactly-once processing, as long as they are reading from / writing to Kafka as the source / sink, their applications cannot actually guarantee that
+            no duplicates will be generated throughout the pipeline.
+
+            Since the 0.11.0.0 release, Kafka has added support to allow its producers to send messages to different topic partitions in a <a href="https://kafka.apache.org/documentation/#semantics">transactional and idempotent manner</a>,
+            and Kafka Streams has hence added the end-to-end exactly-once processing semantics by leveraging these features.
+            More specifically, it guarantees that for any record read from the source Kafka topics, its processing results will be reflected exactly once in the output Kafka topic as well as in the updated processing state.
```

Comment:
Should 'updated processing state' simply be 'state store' ? Or do you mean something else by this?